libraries/AP_Camera/AP_Camera_Backend.cpp

```diff
@@ -241,6 +250,133 @@ void AP_Camera_Backend::send_camera_settings(mavlink_channel_t chan) const
         NaN);               // focusLevel float, percentage from 0 to 100, NaN if unknown
 }
 
+// calculate poi information
+void AP_Camera_Backend::calculate_poi()
+{
+#if AP_TERRAIN_AVAILABLE
+    if (!poi_calculation.sem.take_nonblocking()) {
+        return;
+    }
+    const AP_AHRS &ahrs = AP::ahrs();
+    if (!ahrs.get_location(poi_calculation.loc)) {
+        gcs().send_text(MAV_SEVERITY_ERROR, "POI: vehicle pos unavailable");
+        return;
+    }
+
+    // change vehicle location to ASML
+    poi_calculation.loc.change_alt_frame(Location::AltFrame::ABSOLUTE);
+
+    // retrieve gimbal attitude
+    auto *mount = AP_Mount::get_singleton();
+    Vector3f att;
+    if (!mount->get_attitude_euler(get_mount_instance(),att[0], att[1], att[2]) || isnan(att[1]) || isnan(att[2])) {
```

Comment:
Slightly annoyingly the POI calculation uses Euler (pitch and yaw) while the final MAVLink message requires a quaternion.  You can convert between them or you can take a copy of both here.